osbs/build/spec.py

```diff
@@ -431,3 +431,154 @@ def validate(self):
 
     def __repr__(self):
         return "Spec(%s)" % self.__dict__
+
+
+class BuildUserParams(object):
+    def __init__(self):
+        self.git_uri = BuildParam('git_uri')
+        self.git_ref = BuildParam('git_ref', default=DEFAULT_GIT_REF)
+        self.git_branch = BuildParam('git_branch')
+        self.user = UserParam()
+        self.use_auth = BuildParam("use_auth", allow_none=True)
+        self.platform = BuildParam("platform", allow_none=True)
+        self.component = BuildParam('component')
+        self.name = BuildIDParam()
+        self.build_image = BuildParam('build_image')
+        self.build_imagestream = BuildParam('build_imagestream')
+        self.build_from = BuildParam('build_from')
+        self.proxy = BuildParam("proxy", allow_none=True)
+        self.trigger_imagestreamtag = BuildParam('trigger_imagestreamtag')
+        self.imagestream_name = BuildParam('imagestream_name')
+        self.imagestream_url = BuildParam('imagestream_url')
+        self.imagestream_insecure_registry = BuildParam('imagestream_insecure_registry')
+        self.architecture = BuildParam("architecture")
+        self.platforms = BuildParam("platforms", allow_none=True)
+        self.platform = BuildParam("platform", allow_none=True)
+        self.build_host = BuildParam("build_host")
+        self.release = BuildParam("release", allow_none=True)
+        self.koji_target = BuildParam("koji_target", allow_none=True)
+        self.koji_task_id = BuildParam("koji_task_id", allow_none=True)
+        self.flatpak = BuildParam("flatpak", default=False)
+        self.flatpak_base_image = BuildParam("flatpak_base_image", allow_none=True)
+        self.image_tag = BuildParam("image_tag")
+        self.signing_intent = BuildParam("signing_intent", allow_none=True)
+        self.yum_repourls = BuildParam("yum_repourls")
+        self.compose_ids = BuildParam("compose_ids", allow_none=True)
+        self.reactor_config_map = BuildParam("reactor_config_map", allow_none=True)
+        self.reactor_config_secret = BuildParam("reactor_config_secret", allow_none=True)
```

Comment:
`reactor_config_*` seem more like environment parameters.

Reply:
reactor_config_map is the name of the reactor configuration map. it has to be a user parameter because it's the key needed to retrieve the environment parameters.

reactor_config_secret removed.